.github/workflows/main.yaml

```diff
@@ -0,0 +1,91 @@
+name: CI/CD
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains two jobs named "build" and "container"
+  build:
+    name: Compile and test
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.node-version'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Generate code
+        run: npm run generate
+
+      - name: Compile
+        run: npm run compile
+
+      - name: Lint code
+        run: npm run lint
+
+      - name: Execute tests
+        run: npm test
+
+  container:
+    name: Build container image
```

Comment:
I would change this to a proper name because this job pushes the container image as well in some cases

Reply:
Good point! What do you think about "Build and push the container image". I am aware that the push only occurs under specific circumstances, but I couldn't think of a better name.

An alternative would be to create a third job called "Push the container image" that pushes the image.

Reply:
Maybe the _best_ option could be to split the jobs and call the first one "Build the container image" and the second one "Push the container image", but in this case, I guess you are going to introduce duplicate code, aren't you?
I mean, if you create the push job, that job will contain the login and the extract metadata as well: a copy of what you have done in the build job.

In conclusion, IMHO there are a few possible solutions here:
- keep a single job for build and push, and change the name
- split the jobs -> extracting the common steps
- create a separate repo of the organization containing the jobs to use in order to make the build and the push operation against a container registry

Reply:
I like that option! We can try to share the artifact between jobs!

Reply:
👍 so the best solution is to create a new job that push the container image. I'm going to investigate available options to do that and choose one 💥